File: calc_pi/calc_pi.java
package Ch5;

/**
 * Created by khalid on 10/2/14.
 */
public class Calc_Pi {

    public double calc_pi () {

        long counter = 1;
        double Ret = 0.0;
        long i;
        for(i = 1; counter <= 1000000000L; i = i + 2) {
            if(counter % 2 != 0 ) {
                Ret = Ret + ((double) 4 / i);
            }
            else {
                Ret = Ret - ((double) 4 / i);
            }
            counter += 1;
        }
        return Ret;
    }

    public static void main(String[] args) {

        long tStart = System.nanoTime();

        Calc_Pi pi = new Calc_Pi();

        System.out.println("The calculated value of pi is ...");
        System.out.println(pi.calc_pi());

        System.out.printf("\n%f", (System.nanoTime() - tStart) * .000000001);
    }
}

File: calc_pi/calcPi.py
__author__ = 'khalid'

import time

start = time.time()


def calc_pi():
    counter = 1L
    Ret = 0.0
    i = 1

    while counter <= 1000000000:
        if counter % 2 != 0:
            Ret += 4.0 / i
        else:
            Ret -= 4.0 / i

        counter += 1
        i += 2

    return Ret

print "The calculated value of pi is ... \n"
print calc_pi()

end = time.time()

print "\n"
print end - start

